staff/views/staff.py

```diff
@@ -169,4 +169,53 @@ def slide_delete(request, pk):
         messages.success(request, 'slide has been deleted')
         return redirect('staff:slide_list')
 
+
 # end static info section
+
+
+def create_settings(request):
+    user = UserProfile.objects.get(user__username=request.user.username)
+    allowed = ['superadmin', 'admin']
+    if user.user_type not in allowed:
+        return render(request, 'staff/permissiondenied.html')
+    else:
```

Comment:
You don't need this `else` here as your `if` branch is a return. In general all of this code is super nested, with multiple `if`s sitting inside each other - this is typically much harder to read and keep track of compared to a "return early" style.

Reply:
This is a big picture issue @techlover1 is working to resolve, we're migrating to Django's internal permissions system that will rid of this practice.

Reply:
Sure, but it's a general code problem too. There's lots of if/else branching that I've seen so far (not just related to the perms system). That can all be tidied up to avoid so much logic branching.